File: ecommerce-backend/src/core/order.cpp
#include "order.h"


Order::Order(const std::string& userId, const std::vector<orderItem>& items)
    : id(UUIDGenerator::generateUUID()), userId(userId), items(items) {}

bool Order::saveToDatabase(Database& db) {
    return db.saveOrder(*this);
}

std::string Order::getId() const {
    return id;
}

std::string Order::getUserId() const {
    return userId;
}

std::vector<OrderItem> Order::getItems() const {
    return iteams;
}

double Order::getTotalPrice() const {
    double total = 0;
    for (const auto& item : items) {
        total += product.getPrice();
    }
    return total;
}

File: JAVA/order.java
import java.time.LocalDateTime;
import java.util.List;

public class Order {
    private String orderId;
    private String userId;
    private List<OrderItem> items;
    private LocalDateTime orderDate;
    private double totalPrice;
    private String status;

    public Order(String userId, List<OrderItem> items) {
        this.orderId = UUID.randomUUID().toString();
        this.userId = userId;
        this.items = items;
        this.orderDate = LocalDateTime.now();
        this.totalPrice = calculateTotalPrice();
        this.status = "Pending"; // Default status
    }

    private double calculateTotalPrice() {
        double total = 0;
        for (OrderItem item : items) {
            total += item.getTotalPrice();
        }
        return total;
    }

    public boolean saveToDatabase(Database db) {
        // Implementation to save order to database
    }

    // Getter methods
    public String getOrderId() {
        return orderId;
    }

    public String getUserId() {
        return userId;
    }

    public List<OrderItem> getItems() {
        return items;
    }

    public LocalDateTime getOrderDate() {
        return orderDate;
    }

    public double getTotalPrice() {
        return totalPrice;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}